src/components/GooglePlacesAutocomplete.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import PropTypes from "prop-types";
+import { startCase, debounce } from "lodash";
+import { Input } from "./Forms/Input";
+import { Badge } from "./Badge";
+
+export const GooglePlacesAutocomplete = ({ initialValue, onSelect, urlConfig }) => {
+    const [inputValue, setInputValue] = useState(initialValue || "");
+    const [suggestions, setSuggestions] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [showDropdown, setShowDropdown] = useState(false);
+
+    const dropdownRef = useRef(null);
+    const initialFetchDoneRef = useRef(false);
+
+    const handleSelect = useCallback(
+        (sug) => {
+            setInputValue(sug.description || sug.name || "");
+            setShowDropdown(false);
+            onSelect?.(sug);
+        },
+        [onSelect],
+    );
+
+    const handleInputChange = (e) => {
+        setInputValue(e.target.value);
+        setShowDropdown(true);
+        fetchSuggestions(e.target.value);
+    };
+
+    const fetchSuggestions = useMemo(
+        () =>
+            debounce(async (query, selectFirst = false) => {
+                if (!query.trim()) {
+                    setSuggestions([]);
+                    return;
+                }
+
+                setIsLoading(true);
+                try {
+                    const response = await fetch(`${urlConfig}/searchGooglePlaces?input=${encodeURIComponent(query)}`);
+                    const data = await response.json();
+                    const results = Array.isArray(data) ? data : data.predictions || [];
+                    setSuggestions(results);
+
+                    if (selectFirst && results.length > 0) {
+                        handleSelect(results[0]);
+                    }
+                } catch (error) {
+                    console.error("Error occurred:", error);
+                    setSuggestions([]);
+                } finally {
+                    setIsLoading(false);
+                }
+            }, 300),
+        [urlConfig, handleSelect],
+    );
+
+    useEffect(() => {
+        if (initialValue && !initialFetchDoneRef.current) {
+            initialFetchDoneRef.current = true;
+            fetchSuggestions(initialValue, true);
+        }
+    }, [initialValue, fetchSuggestions]);
+
+    /* eslint-disable no-undef */
+    useEffect(() => {
+        if (typeof document === "undefined") return;
```

Comment:
This is a UI library where document will always be defined. Why do you need this check?